commons/helpers/pods_helper.py

```diff
@@ -206,21 +228,34 @@ def get_deploy_replicaset(self, pod_name):
                       LogicalNode.get_deploy_replicaset.__name__, error)
             return False, error
 
-    def get_num_replicas(self, replicaset):
+    def get_num_replicas(self, set_type, set_name):
         """
-        Helper function to get number of desired, current and ready replicas for given replica set
-        :param replicaset: Name of the replica set
+        Helper function to get number of desired, current and ready replicas for given pod's set
+        type and set name
+        :param set_type: Type of the set (replica set or statefulset)
+        :param set_name: Name of the set
         :return: Bool, str, str, str (Status, Desired replicas, Current replicas, Ready replicas)
         """
         try:
-            log.info("Getting details of replicaset %s", replicaset)
-            cmd = commands.KUBECTL_GET_REPLICASET.format(replicaset)
-            output = self.execute_cmd(cmd=cmd, read_lines=True)
-            log.info("Response: %s", output)
-            output = output[0].split()
-            log.info("Desired replicas: %s \nCurrent replicas: %s \nReady replicas: %s",
-                     output[1], output[2], output[3])
-            return True, output[1], output[2], output[3]
+            log.debug("Set type: %s\n Set name: %s", set_type, set_name)
+            log.info("Getting details of replicaset %s", set_name)
+            if set_type == const.REPLICASET:
+                cmd = commands.KUBECTL_GET_REPLICASET.format(set_name)
+                output = self.execute_cmd(cmd=cmd, read_lines=True)
```

Comment:
Same here. We need to handle IOError exception.

Reply:
same as above